R/args.R

```diff
@@ -147,15 +149,38 @@ SampleArgs <- R6::R6Class(
       self$thin <- thin
       self$max_depth <- max_depth
       self$metric <- metric
-      # self$metric_file <- character()
+      self$inv_metric <- inv_metric
+      if(!is.null(inv_metric)) {
+        if(!is.null(metric_file)) {
+          stop("Only one of inv_metric and metric_file can be non-NULL")
+        }
+
+        # wrap inv_metric in list if not in one
+        if(!is.list(inv_metric)) {
+          inv_metric = list(inv_metric)
+        }
+
+        # write all inv_metrics to disk
+        inv_metric_paths <- c()
+        for(metric_i in 1:length(inv_metric)) {
+          inv_metric_path <-
+            file.path(
+              cmdstan_tempdir(),
+              paste0("inv_metric-", metric_i, ".json")
+            )
+          write_stan_json(list(inv_metric = inv_metric[[metric_i]]), inv_metric_path)
+          inv_metric_paths <- c(inv_metric_paths, inv_metric_path)
+        }
```

Comment:
This is nitpicking, but I think it's a bit cleaner if you create all the files together and then use a shorter loop instead of accumulating up the vector of paths. For example:

```r
# also incorporating the addition of `tempfile` mentioned above
inv_metric_paths <-
  tempfile(
    pattern = paste0("inv_metric-", 1:length(inv_metric), "-"),
    tmpdir = cmdstan_tempdir(),
    fileext = ".json"
  )
for (i in seq_along(inv_metric_paths)) {
  write_stan_json(list(inv_metric = inv_metric[[i]]), inv_metric_paths[i])
}
```

It's nitpicky so if you disagree it's fine to leave as is!